gammapy/time/plot.py

```diff
@@ -2,31 +2,126 @@
+    fermi_cat = fetch_fermi_catalog('3FGL')
+
+    try:
+        catalog_index = np.where(fermi_cat[1].data['Source_Name'] == name_3FGL)[0][0]
+    except IndexError:
+        print("Could not find " + name_3FGL + " in the 3GL catalog.")
+        return
+
+    time_index_start = np.where(fermi_cat[3].data['Hist_Start'] >= fermi_met_start)[0][0]
+
+    # The final entry is the end of the last bin, so no off by one error
+    time_index_end = np.where(fermi_cat[3].data['Hist_Start'] <= fermi_met_end)[0][-1] + 1
+
+    time_start = fermi_cat[3].data['Hist_Start'][time_index_start: time_index_end]
+    time_end = np.roll(time_start, -1)
+
+    time_diff = time_end - time_start
+
+    # Trim because there is one more bin edge than there is bin mid point
+    time_diff = time_diff[0:-1]
+
+    # Midpoints of each bin.
+    time_mid = time_start[0:-1] + time_diff
+
+    flux_history = fermi_cat[1].data[catalog_index]['Flux_History'][time_index_start: time_index_end]
+
+    flux_history_lower_bound = fermi_cat[1].data[catalog_index]['Unc_Flux_History'][time_index_start: time_index_end, 0]
+    flux_history_upper_bound = fermi_cat[1].data[catalog_index]['Unc_Flux_History'][time_index_start: time_index_end, 1]
+    flux_history_lower_bound = abs(flux_history_lower_bound)
+
+    # Change bins with no flux value from nan to zero.
+    for i in range(0, np.size(flux_history_lower_bound)):
+        if math.isnan(flux_history_lower_bound[i]):
+            flux_history_lower_bound[i] = 0
+
+    # Create array of upper limits where no lower bound was recorded.
+    upper_lims_x = np.copy(time_mid)
+
+    upper_lims_y_start = np.copy(flux_history_upper_bound)
+
+    upper_lims_y = np.copy(flux_history)
+
+    upper_lims_x = \
+        upper_lims_x[np.where(flux_history_lower_bound <= 0)]
+
+    upper_lims_y_start = \
+        upper_lims_y_start[np.where(flux_history_lower_bound <= 0)]
+
+    upper_lims_y = \
+        upper_lims_y[np.where(flux_history_lower_bound <= 0)]
+
+    upper_lims_y_end = np.copy(upper_lims_y_start)
+
+    # Scale the size of the arrow to an arbitrary value.
+    upper_lims_y_end *= -0.3
+
+    # Create an array of data points where a lower bound was recorded.
+    time_mid = \
+        time_mid[np.where((flux_history_lower_bound) > 0)]
+
+    time_diff = \
+        time_diff[np.where((flux_history_lower_bound) > 0)]
+
+    flux_history = \
+        flux_history[np.where((flux_history_lower_bound) > 0)]
 
-    if ax is None:
-        ax = plt.gcf()
+    flux_history_upper_bound = \
+        flux_history_upper_bound[np.where((flux_history_lower_bound) > 0)]
 
-    td = time[1:] - time[:-1]
+    flux_history_lower_bound = \
+        flux_history_lower_bound[np.where((flux_history_lower_bound) > 0)]
 
-    # TODO: implement!
-    raise NotImplementedError
+    # Plot data points and upper limits.
+    plt.errorbar(time_mid, flux_history, yerr=(flux_history_lower_bound, flux_history_upper_bound), \
```

Comment:
No need for `\` at the end of the lines. The fact that the `plt.errorbar()` function call has a start and stop parenthesis means it's OK to use multiple lines for the function arguments.